pl_bolts/datamodules/base_datamodule.py

```diff
@@ -0,0 +1,120 @@
+import os
+from typing import Optional, Tuple
+
+import torch
+from pytorch_lightning import LightningDataModule
+from torch.utils.data import DataLoader, random_split
+
+from pl_bolts.utils.warnings import warn_missing_pkg
+
+try:
+    from torchvision import transforms as transform_lib
+except ModuleNotFoundError:
+    warn_missing_pkg("torchvision")  # pragma: no-cover
+    _TORCHVISION_AVAILABLE = False
+else:
+    _TORCHVISION_AVAILABLE = True
+
+
+class BaseDataModule(LightningDataModule):
+
+    extra_args = {}
+
+    def __init__(
+        self,
+        dataset_cls,
+        dims: Tuple[int, int, int],
+        data_dir: Optional[str] = None,
+        val_split: int = 5000,
+        num_workers: int = 16,
+        normalize: bool = False,
+        seed: int = 42,
+        batch_size: int = 32,
```

Comment:
I like the idea of `eval_batch_size` and `train_batch_size` being separate

Reply:
This would break a lot of tests. Should I still do it?

Reply:
Let's move this change to separate PR?
in theory, how much the train and valid batch size can have an effect?